src/engine/Join.cpp

```diff
@@ -819,13 +790,13 @@ ProtoResult Join::computeResultForIndexScanAndIdTable(
             !idTable.empty() && idTable.at(0, joinColTable).isUndefined();
         std::optional<std::shared_ptr<const Result>> indexScanResult =
             std::nullopt;
-        using FirstColView = ad_utility::IdTableAndFirstCol<IdTable>;
-        using GenWithDetails =
-            cppcoro::generator<FirstColView,
-                               CompressedRelationReader::LazyScanMetadata>;
+        using GenWithDetails = decltype(convertGenerator(
+            std::declval<Permutation::IdTableGenerator>()));
         auto rightBlocks = [&scan, idTableHasUndef, &permutationIdTable,
                             &indexScanResult]()
-            -> std::variant<cppcoro::generator<FirstColView>, GenWithDetails> {
+            -> std::variant<decltype(convertGenerator(
+                                std::declval<Result::Generator>())),
```

Comment:
Also give this alias a name `(GenWithoutDetails)` or something, to make the code more symmetric and self-speaking.